### ISSUE TITLE
_.merge modifies source objects

### ISSUE THREAD

**Issue by ellis** [lodash/lodash#1801]
I just upgraded to lodash 4.0.0, and encountered a bizarre error in my program which I finally tracked down to a bug in _.merge.  Here's a minimal example I put in a file named `temp.js`:

```{JavaScript}
var _ = require('lodash');
var a = {predicates: [{a: 1}]};
var b = {predicates: [{b: 2}]};
var c = _.merge({}, a, b);
console.log("a.predicates: "+JSON.stringify(a.predicates));
console.log("b.predicates: "+JSON.stringify(b.predicates));
console.log("c.predicates: "+JSON.stringify(c.predicates));
```

Running `node temp.js` produces the following output:

```
a.predicates: [{"a":1,"b":2}]
b.predicates: [{"b":2}]
c.predicates: [{"a":1,"b":2}]
```

I would instead expect the value of `a.predicates` to be unchanged.  What do you think?

**Comment by jdalton** [comment]
I've got a fix for this and #1794 coming shortly. Just have to write the unit tests (which you've provided) :+1:

**Comment by ellis** [comment]
Great, thanks!
